base_report_to_printer/report.py

```diff
@@ -53,6 +71,9 @@ def get_pdf(self, cr, uid, ids, report_name, html=None,
         report = self._get_report_from_name(cr, uid, report_name)
         behaviour = report.behaviour()[report.id]
         printer = behaviour['printer']
-        if behaviour['action'] == 'server' and printer and document:
+        can_print_report = self._can_print_report(cr, uid, ids,
+                                                  behaviour, printer, document,
+                                                  context=context)
+        if can_print_report:
```

Comment:
@guewen my point was more about mro trouble caused by dependency issues on parallel addons that may cause multiple call to printer. But ok lets see if the bug happens and we will advise then